### PR TITLE
Hotfix/1857 email recipients revisited

### PR DESCRIPTION
**Pull request by richard-jones**
set journal contact and account holder where present as the recipients for in progress and editor assigned emails

### CODE REVIEW

File: portality/formcontext/emails.py [hunk 1]
```diff
@@ -200,25 +213,36 @@ def send_publisher_inprogress_email(application):
...
+    send_list = [(contact_name, contact_email,
+                  Messages.SENT_JOURNAL_CONTACT_IN_PROGRESS_EMAIL,
+                  Messages.NOT_SENT_JOURNAL_CONTACT_IN_PROGRESS_EMAIL)]
 
-    # This is to the publisher contact on the application
-    publisher_name = owner.name
-    publisher_email = owner.email
+    owner = models.Account.pull(application.owner)
+    if owner is not None:
+        send_list.append((owner.name, owner.email,
```
Comment (Steven-Eardley) [inline portality/formcontext/emails.py]: I find this a bit of a weird data structure to use for this - you have a tuple of name, email, and 2 messages - but these are unlabelled and undocumented.

File: portality/formcontext/formcontext.py [hunk 2]
```diff
@@ -1056,15 +1052,11 @@ def finalise(self):
...
-                except app_email.EmailException:
-                    magic = str(uuid.uuid1())
-                    self.add_alert('Hm, sending the ready status to publisher email didn\'t work. Please quote this magic number when reporting the issue: ' + magic + ' . Thank you!')
-                    app.logger.exception('Error sending ready status email to publisher - ' + magic)
+                alerts = emails.send_publisher_inprogress_email(self.target)
+                for alert in alerts:
+                    self.add_alert(alert)
             else:
-                self.add_alert('Did not send email to Journal Contact about the status change, as publisher emails are disabled.')
+                self.add_alert('Did not send email to Owner or Journal Contact about the status change, as publisher emails are disabled.')
```
Comment (Steven-Eardley) [inline portality/formcontext/formcontext.py]: should this be a Message?

File: portality/formcontext/emails.py (hunk 1 above)
Comment (richard-jones) [reply portality/formcontext/emails.py]: Fair enough, switching to using a dict for tracking the values.

File: portality/formcontext/formcontext.py (hunk 2 above)
Comment (richard-jones) [reply portality/formcontext/formcontext.py]: Yeah, should be.  There's a lot in that file that should be, but I'll stick to only moving stuff out as we touch it.  So, I've moved this one :)

**Comment by Steven-Eardley** [comment]
This one is ready to go. Can't release until the article duplication migration is complete - in a few hours.

**Pull request closed without merging**